reportportal_client/service.py

```diff
@@ -249,6 +249,40 @@ def finish_launch(self, end_time, status=None, **kwargs):
         logger.debug("finish_launch - ID: %s", self.launch_id)
         return _get_msg(r)
 
+    def get_launch_info(self):
+        """Get the current launch information.
+
+        :return dict: launch information
+        """
+        url = uri_join(self.base_url_v1, "launch/uuid", self.launch_id)
+        launch_info = _get_json(self.session.get(
+            url=url, verify=self.verify_ssl))
+        logger.debug("get_launch_info - ID: %s", self.launch_id)
+        logger.debug("get_launch_info - Launch info: %s", launch_info)
+        return launch_info
+
+    def get_launch_ui_id(self):
+        """Get UI ID of the current launch.
+
+        :return str: UI ID of the given launch.
+        """
+        launch_info = self.get_launch_info()
+        if "id" in launch_info:
+            return launch_info["id"]
+
+        raise ResponseError("No UI ID in response {0}.".format(launch_info))
```

Comment:
Is it really possible that the returned JSON won't include the "id" key?

Reply:
Hi,

Thanks for the review!
 I don`t know. Not familiar with RP internals yet. I assumed one of the fantastic scenarios and wanted to make the error message more descriptive:  
- if requested before launch created and we got the `UI ID`, AFAIU it is the DB id? 
- if in future this field will be renamed to e.g. "UI_ID"

But since the question raised... Ok, will remove this extra code.

Of course in all the cases we will **KeyError**  that would be enough.

Reply:
You can get this error when you are requesting a launch ID and that launch has not been processed at the RP backend. You can address this if you want.